### ISSUE TITLE
Cannot change keyboard shortcuts in Firefox on macOS

### ISSUE THREAD

**Issue by beingmrkenny** [keepassxreboot/keepassxc-browser#777]
## Expected Behavior

1. Visit /options/shortcuts.html in keePass options
2. Click input field that contains the shortcut (e.g. MacCtrl+Shift+u)
3. Press new shortcut
4. Shortcut is saved

## Current Behavior

Input field remains empty, or is briefly filled with a character like È, no shortcut is recorded, and no new shortcuts are saved.

## Possible Solution

Seems like a JS problem, possibly to do with not recognising which key has been pressed, but I haven't been able to look into it in depth.

## Steps to Reproduce (for bugs)

See expected behaviour

## Debug info

No error messages returned in console

KeePassXC-Browser Version: 1.5.4
KeePassXC Version: 2.5.3 
Operating system: macOS 10.14.6
Browser: Firefox 73.0

**Comment by varjolintu** [comment]
I'm actually having the same problem sometimes, because of my exotic keyboard layout. Let's see if it's possible to ignore the modifier keys for the last character.

**Comment by varjolintu** [comment]
This probably because of the keyboard layout, and I'm not sure if there's anything we can do about it. What shortcuts gives you È and what's your layout?

**Comment by beingmrkenny** [comment]
British layout. Checking this again, I only get accented characters etc with the alt key, which I wouldn't use for a shortcut. Using the ctrl key works as expected.

I would like to use the cmd key, which on macos is used like ctrl is on other systems. Trying to use the command key causes the bugs detailed above.

Is it possible to use the cmd key?

**Comment by varjolintu** [comment]
AFAIK with Firefox using Cmd with extension keyboard shortcuts is not possible.

**Comment by beingmrkenny** [comment]
1Password seems to manage it, but it might be registering the keypress via the app that runs outside of the browser.

**Comment by varjolintu** [comment]
Only keyboard shortcut with 1PasswordX I see is Cmd+Period which is used as a "Activate extension". It's a different thing.

**Comment by beingmrkenny** [comment]
There's cmd+/, which has the same effect as "fill with username and password". This is 1Password that comes with the app from the app store, presumably 1PasswordX works differently.

If I clone the keepasscx-browser to take a look locally, will it still work with my local database?

**Comment by varjolintu** [comment]
Yes, you can clone the repo and load it as a temporary extension. No building or anything special needed.

**Comment by cmroanirgo** [comment]
I have this problem too. Actually, I can't seem to type anything in those shortcut boxes. Even "}", it just flashes and then empties.

FFox v73
OSX 10.13.6 International Keyboard (The top row is ~!@#$%^&*()_+ which is above the keys `1234567890-=    )
KPXC: 2.5.3
KPXC-Browser: 1.5.4
I can concur that 1Password6 & 7 can autofill using Cmd+\ It works well.

No idea how to type the È other than typing E (shift+E) and holding for a few secs and then use the mouse.

**Comment by varjolintu** [comment]
We are talking about browser extension shortcuts here. As you can [see](https://support.1password.com/keyboard-shortcuts/) these shortcuts are related to the desktop application.

**Comment by cmroanirgo** [comment]
This is about browser extension (including 1Password). Even your iink clearly shows that it's the browser extension that uses Cmd+\\ . 

> Command‑Backslash (\) | Fill Login on current web page. Requires the 1Password extension.




I disabled the global auto type (in the KPXC), because it didn't work in firefox and tried to use the same hotkey Cmd+] in the browser instead. Result: fail (& failed with any attempt at different shortcut).

**Comment by varjolintu** [comment]
> This is about browser extension (including 1Password). Even your iink clearly shows that it's the browser extension that uses Cmd+\ .

This is not about the browser extension. The page clearly says those are Global keyboard shortcuts for the application itself. Probably they just pass the command from the desktop application to the extension which does the filling.

> I disabled the global auto type (in the KPXC), because it didn't work in firefox and tried to use the same hotkey Cmd+] in the browser instead. Result: fail (& failed with any attempt at different shortcut). It's interesting about the keyboard restrictions in Web Extensions...

In short, Web Extensions must also use modifier keys, so only `Ctrl/Alt` or `Ctrl+Shift/Alt+Shift` are accepted with the additional key.

> EDIT: There's a project (https://github.com/philc/vimium) that allows vim style keyboard entry in the browser, clearly bypassing the WebExtensions 'commands'. From a rough glance it seems that a content script is injected and the keyboard listening is done there.
> There is also a firefox bug (https://bugzilla.mozilla.org/show_bug.cgi?id=1215061) that mentions it's possible that the popup _also_ has access to all keyboard strokes (but getting back to the content page is tricky).

Of course it's possible to inject identical method, but at this point we are not overriding browser's own shortcuts.

**Comment by aism** [comment]
Same goes for Chrome indeed - tried to set the Command+\ as I had in 1Password I've escaped from (so I suggest to rename the title of this Issue to mention the Chrome as well).

It would be cool to have more flexible shortcut options as I dunno when/if I'll be able to override the many years of muscle memory...
